gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcDirectStreamController.java

```diff
@@ -110,10 +109,10 @@ void startBidi() {
   private void startCommon() {
     responseObserver.onStart(this);
 
-    this.hasStarted = true;
-
     clientCall.start(new ResponseObserverAdapter(), new Metadata());
 
+    this.hasStarted = true;
```

Comment:
Why this line is moved from before `clientCall.start` to after it? I don't know the exact use case of `startCommon`, but it could be an issue if `clientCall.start` is taking some time to complete and `startCommon` is being called another time.

Reply:
Thanks, it makes sense.